### PR TITLE
adding Serply as another web search provider

### PR DESCRIPTION
**Pull request by googio**
# Pull Request Checklist

### Note to first-time contributors: Please open a discussion post in [Discussions](https://github.com/open-webui/open-webui/discussions) and describe your changes before submitting a pull request.

**Before submitting, make sure you've checked the following:**

- [X] **Target branch:** Please verify that the pull request targets the `dev` branch.
- [x] **Description:** Provide a concise description of the changes made in this pull request.
- [x] **Changelog:** Ensure a changelog entry following the format of [Keep a Changelog](https://keepachangelog.com/) is added at the bottom of the PR description.
- [ ] **Documentation:** Have you updated relevant documentation [Open WebUI Docs](https://github.com/open-webui/docs), or other documentation sources?
- [x] **Dependencies:** Are there any new dependencies? Have you updated the dependency versions in the documentation?
- [x] **Testing:** Have you written and run sufficient tests for validating the changes?
- [ ] **Code review:** Have you performed a self-review of your code, addressing any coding standard issues and ensuring adherence to the project's coding standards?
- [ ] **Prefix:** To cleary categorize this pull request, prefix the pull request title, using one of the following:
  - **BREAKING CHANGE**: Significant changes that may affect compatibility
  - **build**: Changes that affect the build system or external dependencies
  - **ci**: Changes to our continuous integration processes or workflows
  - **chore**: Refactor, cleanup, or other non-functional code changes
  - **docs**: Documentation update or addition
  - **feat**: Introduces a new feature or enhancement to the codebase
  - **fix**: Bug fix or error correction
  - **i18n**: Internationalization or localization changes
  - **perf**: Performance improvement
  - **refactor**: Code restructuring for better maintainability, readability, or scalability
  - **style**: Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc.)
  - **test**: Adding missing tests or correcting existing tests
  - **WIP**: Work in progress, a temporary label for incomplete or ongoing work

# Changelog Entry

### Description

Added function to perform web search with Serply.io because Serply supports Google, News, and Jobs search. It will be beneficial to be able to perform News and Job search in the future. 

### Added

- Added `SERPLY_API_KEY` API Key for config
- Added `serply` as an option for `webSearchEngines`
- Added `backend/apps/rag/search/serply.py` which is the function for calling [Serply](https://serply.io) Google Search API
 * Builds the query payload
 * Passes in SERPLY_API_KEY into the headers
- Updated `/config/update` to set `SERPLY_API_KEY` in the config
- Added serply testdata

### Changed

- Updated README.md to `Serply.io` as an option for Web Search

### Additional Information

- Enhancement, added more web search providers
  - [[Reference any related issues, commits, or other relevant information]](https://github.com/open-webui/open-webui/issues/2957#issue-2342613119)

### Screenshots or Videos

- [Attach any relevant screenshots or videos demonstrating the changes]

### CODE REVIEW

File: src/lib/components/documents/Settings/WebParams.svelte
Comment (tjbck) [inline src/lib/components/documents/Settings/WebParams.svelte:1]: This files' deprecated.

**Comment by tjbck** [comment]
Closing for now, but feel free to reopen with just the backend/frontend changes!

File: src/lib/components/documents/Settings/WebParams.svelte
Comment (googio) [reply src/lib/components/documents/Settings/WebParams.svelte:1]: So where is are the new config being changed?

Comment (tjbck) [reply src/lib/components/documents/Settings/WebParams.svelte:1]: WebSearch in Admin Settings!

Comment (googio) [reply src/lib/components/documents/Settings/WebParams.svelte:1]: Cool thanks. I'll update that.

**Pull request closed without merging**